### ISSUE TITLE
Remove/Destroy containers - doesn't properly handle children

### ISSUE THREAD

**Issue by jarvisteach** [jarvisteach/appJar#75]
If you call removeLabelFrame()  to destroySubWindow()

The container (and its children) is destroyed fine, however the widgets it holds are left in the config dictionaries.

This generally doesn't cause a problem - but if they are later accessed (e.g. in a callback loop such as animation, via change language, or just be a standard user call) it's not possible to call functions on them as they've been destroyed.

So, when a container is removed/destroyed - need some way of removing all of its children form the config arrays...

**Comment by jarvisteach** [comment]
Added attempt to cleanse widgets when a SubWindow is destroyed.

Convoluted, have to hunt down the widget type, find its ID, use that to get its dictionary, then hunt through the dictionary to find the widget.

Not all widgets are in the ID dictionary (WIDGETS) - so had to be done manually.

Will raise a separate issue, to discuss changing how this is all managed.

**Comment by jarvisteach** [comment]
Putting this issue on hold, until after #147 is resolved.

When a widget is removed, it needs to go through the same process as above.

This also needs to check variables are removed, event loops are removed, etc.

**Comment by jcolom64** [comment]
From what I can tell, this will also impact removeTabbedFrame. If you create a TabbedFrame with one Tab named "A", then remove the frame by calling removeTabbedFrame, the frame and widgets are removed. But then, if you try to recreate the TabbedFrame with the same Tab "A" you will get the exception:  Duplicate tabName: A. As far as I can tell, the problem is that the widgetStore still contain Tab "A".

**Comment by jarvisteach** [comment]
Closing as covered by issue #462
